### PR TITLE
166840932 List users functionality

### PR DESCRIPTION
**Pull request by Onyimatics**
### What does this PR do?
- List users functionality

#### Description of Task to be completed?
- create a database locally
- create a user controller
- create the corresponding API endpoint
- document the endpoint with swagger

#### How should this be manually tested?
- clone the repo
- run npm install
- start the server with npm run start:dev
- test on postman

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#166840932](https://www.pivotaltracker.com/story/show/166840932)

#### Screenshots (if appropriate)
Screenshot 2019-07-03 at 4 00 58 AM

#### Questions:
- N/A

### CODE REVIEW

**Review by kechyy — COMMENTED** [review]
Good work @Onyimatics . Looks good to me.

**Review by kechyy — APPROVED** [review]

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  static async listUsers(req, res, next) {
+    try {
+      const users = await db.User.findAll({
+        attributes: ['id', 'username', 'email', 'bio', 'avatar'],
+        where: {
+          status: 'active'
+        }
+      });
+      return res.status(200).json({
+        status: 200,
```
Comment (segunolalive) [inline src/controllers/user.controller.js]: please remove the status key and the message

**Review by segunolalive — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.js (hunk 1 above)
Comment (Onyimatics) [reply src/controllers/user.controller.js]: All right, thank you. I will remove them.

File: src/test/user.spec.js [hunk 2]
```diff
@@ -277,3 +277,46 @@ describe('GET api/v1/users', () => {
...
+      });
+  });
+  it('should successfully get a list of all users', done => {
+    const userUrl = '/api/v1/auth/users';
+    chai
+      .request(app)
+      .get(userUrl)
+      .set('Authorization', superAdminToken)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
```
Comment (segunolalive) [inline src/test/user.spec.js]: List of users in not demonstrably tested here.

File: src/test/user.spec.js [hunk 3]
```diff
@@ -277,3 +277,46 @@ describe('GET api/v1/users', () => {
...
+        done();
+      });
+  });
+  it('should not allow list of all users', done => {
```
Comment (segunolalive) [inline src/test/user.spec.js]: this route should not be protected.

**Review by segunolalive — CHANGES_REQUESTED** [review]

**Pull request merged**